xds/src/main/java/io/grpc/xds/internal/security/certprovider/CertProviderClientSslContextProvider.java

```diff
@@ -56,12 +54,14 @@ final class CertProviderClientSslContextProvider extends CertProviderSslContextP
   protected final SslContextBuilder getSslContextBuilder(
           CertificateValidationContext certificateValidationContextdationContext)
       throws CertStoreException {
-    SslContextBuilder sslContextBuilder =
-        GrpcSslContexts.forClient()
-            .trustManager(
-                new XdsTrustManagerFactory(
-                    savedTrustedRoots.toArray(new X509Certificate[0]),
-                    certificateValidationContextdationContext));
+    SslContextBuilder sslContextBuilder = GrpcSslContexts.forClient();
+    if (rootCertInstance != null
+        && !tlsContext.getCommonTlsContext().getValidationContext().hasSystemRootCerts()) {
```

Comment:
I don't think this is right. `ca_certificate_provider_instance` is supposed to be preferred over `system_root_certs`. From the gRFC:

> We have added a system_root_certs field to the xDS CertificateValidationContext message. In the gRPC client, if this field is present and the ca_certificate_provider_instance field is unset, system root certificates will be used for validation.

Reply:
The introduction of the system_root_certs field then seems redundant as the presence or absence of ca_certificate_provider_instance itself will decide whether the system root certs get used. For using system root certs we have to _do nothing_ here w.r.t. setting trust manager in the SSLContextBuilder.

Reply:
I have removed the 2nd condition from now with the understanding that the new boolean field system_root_certs may be redundant for Java but other implementations may still have a use for them.

Reply:
ClientSslContextProviderFactory has a check asserting that at least on of the fields is set. So it looks like that verification is interacting with the condition here. I'm not at all wild about how all this is plumbed; we should convert the tlsContext/CommonTlsContext into our own internal data model. But this is fine for now.